plonky3/src/params/poseidon2/goldilocks/mod.rs

```diff
@@ -0,0 +1,23 @@
+#[cfg(not(all(target_os = "zkvm", target_arch = "riscv32")))]
+mod software_impl;
+#[cfg(not(all(target_os = "zkvm", target_arch = "riscv32")))]
+pub use software_impl::*;
+
+#[cfg(all(target_os = "zkvm", target_arch = "riscv32"))]
+mod powdr_accel_impl;
+#[cfg(all(target_os = "zkvm", target_arch = "riscv32"))]
```

Comment:
I think we should also add a feature `powdr` to the crate and these checks, in case we want to run inside powdr but with software impl. Wdyt?

Reply:
I prefer with a `powdr` feature, but I did the way you told me to do.

Reply:
maybe we should just use the 3? I mean, the Cargo.toml powdr-riscv-runtime dep needs the target anyway, we could be restrictive here too

Reply:
3 what? 3 conditions in the `all`?

Reply:
yea, the current 2 in the PR plus the feature

Reply:
I don't see it would be better than just using the feature.

If compiling to powdr, both with and without the feature would be valid.

But if not, the feature should not be used and the user should get an error if attempted.

Reply:
Exactly. I'm fine with only using the feature, but then we allow the user to try to compile to native using the feature, which will fail ofc but will be ugly. If we use the target as well the user actively needs to be using powdr to go that path

Reply:
Imagine the user builds for x86-64 and sets the feature. What do you want to happen?

I want a compilation error, and that is what will happen if we only the use the feature.

If we use the 3 conditions, the path is not taken and the feature is silently ignored.

Reply:
Ok, this is not possible unless we accept to break `--all-features`, which I think we don't. The fact that the feature could not be enabled at x86-64 broke our CI, because we build powdr with `--all-features`. I changed for the feature to be silently ignored.